configure.ac

```diff
@@ -4513,26 +4513,30 @@ AC_ARG_WITH(lazy_gc_thread_creation, [  --with-lazy-gc-thread-creation=yes|no
 	fi
 ], [with_lazy_gc_thread_creation=no])
 
-AC_ARG_WITH(cooperative_gc, [  --with-cooperative-gc=yes|no      Enable cooperative stop-the-world garbage collection (sgen only) (defaults to no)], [], [with_cooperative_gc=default])
+AC_ARG_WITH(cooperative_gc,      [  --with-cooperative-gc=yes|no           Enable cooperative stop-the-world garbage collection (sgen only) (defaults to no)], [], [with_cooperative_gc=default])
+AC_ARG_WITH(cooperative_suspend, [  --with-cooperative-suspend=yes|no      Enable cooperative stop-the-world garbage collection (sgen only) (defaults to no)], [], [with_cooperative_suspend=default])
 
-if test x$with_cooperative_gc = xdefault; then
-	with_cooperative_gc=$with_cooperative_gc_default
+if test x$with_cooperative_suspend = xdefault -a test x$with_cooperative_gc != xdefault; then
+	with_cooperative_suspend=$with_cooperative_gc
```

Comment:
if I'm reading the logic correctly we never assign the actual value to `with_cooperative_gc` (only `default` in case the option is omitted)

Reply:
We use `with_cooperative_gc` only if it's none default and `with_cooperative_suspend` is default, ie we specified `--with-cooperative-gc=yes|no` but not `--with-cooperative-suspend=yes|no`. If we specified `--with-cooperative-suspend=yes|no`, then we discard `--with-cooperative-gc`.

Reply:
Ah yeah I misremembered that `AC_ARG_WITH ()` defaults the `action-if-present` to setting the variable to the value. I thought we'd need to set it manually.